src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,82 @@ class C1<T>
             Assert.Equal(c1.MangleName, c1r.MangleName);
             Assert.Equal(c1.MetadataName, c1r.MetadataName);
         }
+
+        [Fact]
+        public void RetargetingFunctionPointerType_01()
+        {
+            string source = @"
+unsafe class C
+{
+    delegate*<out string, ref readonly int> M(delegate* cdecl<void> param) => throw null;
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview, options: TestOptions.UnsafeReleaseDll);
+            comp.VerifyDiagnostics();
+
+            var sourceModule = comp.SourceModule;
+            var sourceAssembly = (SourceAssemblySymbol)sourceModule.ContainingAssembly;
+            var sourceNamespace = sourceModule.GlobalNamespace;
+
+            var retargetingAssembly = new RetargetingAssemblySymbol(sourceAssembly, isLinked: false);
```

Comment:
>var retargetingAssembly = new RetargetingAssemblySymbol(sourceAssembly, isLinked: false); [](start = 12, length = 89)

Let's not do this. Let's get retargeting assembly created by the system and than assert our expectations.